python/TestHarness/util.py

```diff
@@ -519,23 +519,23 @@ def getLibtorchVersion(moose_dir):
     return major_version.pop() + '.' + minor_version.pop()
 
 def checkLogicVersionSingle(checks, iversion, package):
-    logic, version = re.search(r'(.*?)(\d\S+)', iversion).groups()
+    logic, version = re.search(r'(.*?)\s*(\d\S+)', iversion).groups()
     if logic == '' or logic == '=':
         if version == checks[package]:
             return True
         else:
             return False
 
-    # Logical match
-    if logic == '>' and list(map(int, checks[package].split("."))) > list(map(int, version.split("."))):
-        return True
-    elif logic == '>=' and list(map(int, checks[package].split("."))) >= list(map(int, version.split("."))):
-        return True
-    elif logic == '<' and list(map(int, checks[package].split("."))) < list(map(int, version.split("."))):
-        return True
-    elif logic == '<=' and list(map(int, checks[package].split("."))) <= list(map(int, version.split("."))):
-        return True
+    from operator import lt, gt, le, ge
+    ops = {
+      '<': lt,
+      '>': gt,
+      '<=': le,
+      '>=': ge,
+    }
 
+    if ops[logic]([int(x) for x in checks[package].split(".")], [int(x) for x in version.split(".")]):
```

Comment:
```suggestion
    if ops[logic](checks[package].split("."), version.split(".")):
```
does this work?

Reply:
split() will leave a string version of the number, so you have to convert it to int, and the fastest way to do it is with the list comprehension

Reply:
but python can compare strings right?

Reply:
it can but > and < will compare whether the string has more characters in it, not whether the numbers in the string are bigger or smaller

Reply:
if you compare '3' < '4' it'll say False because 3 and 4 are the same length, which isn't helpful for determining what version number we are on

Reply:
seems that case actually works as expected
```
>>> a3 = "3.10"
>>> a2 = "3.11.7"
>>> ge(a2.split('.'), a3.split('.'))
True
>>> a3 = "3.14"
>>> ge(a2.split('.'), a3.split('.'))
False
```

But Cody's is the problem
```
>>> ge(2, 10)
False
>>> ge("2", "10")
True
```

Reply:
Ok for the "3" and "4" case

string conversions not as helpful as expected!